app/models/vm/operations/lifecycle.rb

```diff
@@ -7,14 +7,18 @@ module Vm::Operations::Lifecycle
     end
 
     supports :migrate do
-      if blank? || orphaned? || archived?
-        unsupported_reason_add(:migrate, "Migrate operation in not supported.")
+      unsupported_reason_add(:migrate, 'Migrate operation is not supported') unless supports_lifecycle_operation?
+    end
+
+    supports :clone do
+      unless supports_lifecycle_operation? || cloneable?
+        unsupported_reason_add(:clone, 'Clone operation is not supported')
       end
     end
   end
 
-  def validate_clone
-    {:available => self.cloneable? && !(self.blank? || self.orphaned? || self.archived?), :message => nil}
+  def supports_lifecycle_operation?
```

Comment:
1. dont add `supports_` methods by defining methods. They should all be added via the `SupportsFeatureMixin`
2. I prefer to have the checks inlined, although it might look like code duplication, but in this case the benefit of having the check in one place outweighs it.

so just add the `supports :clone` block just as the `:migrate` block